### DISCUSSION TITLE
Newbie, can anyone point me in the right direction? Thanks in advance! :)

### DISCUSSION THREAD

**Discussion by Flamingo-verse** [community/community#21524] [Other Feature Feedback, Questions, & Ideas]
<p>First up I just want to say thanks in advance for reading and/or helping out! I’m completely new to coding, I’m sure I look like a fish out of water to you guys, but I appriciate you for even taking the time to read… And I hope I’m posting this in the right area.</p>
<p>I’m studying in the veterinary feild and the idea is I would like to create a customisable learning tool/program to help remembering terminology and procedures, by imputting my own library of questions and answers, maybe with images as well. Questions that will be pulled at random that I’ll have to give the correct answer to, or maybe pick from muliple choice pulled at random from the answer library. With the goal being to use this as a way to strengthen my memory of terminology and procedures.</p>
<p>So I’d like to know what language would be best to learn for that, what programs I might need, any sites I should visit etc. I don’t know the first thing about coding, litteraly ziltch. I’m totally fine with learning it though, it’s just I have a very busy day to day schedual so I dont want to waste what little free time I have learning one language only to find out I should have been learning a different one. That’s why I thought it would be wise to reach out and ask those who know a lot more about it, just to point me in the right direction. I feel as far as programs go this should be fairly straight forward but again, I wouldnt even know where to start or which language would be best, and google searching only led me to things like custom “FB quizes”, not what I was after.</p>
<p>I’d be greatly appreciative for any help on this, I’m fully prepared to learn and do it all myself, would just like someone to point me where to start, so I know I’m not going down the wrong direction, thanks!</p>

**Comment by fire-eggs** [comment, marked as answer]
<p>You’ll want to refine your goals a little bit. I’m assuming you want a desktop program [i.e. not mobile, not web]. Windows, OSX or Linux?</p>
<p>I personally suggest Python but Javascript is also a good “starter” language. Python works on all platforms; I have no experience in Javascript.</p>
<p>For learning Python I suggest a number of tutorials or intro, especially “Learn Python the Hard Way”.</p>
<p>Instead of “do it all myself”, I’d take advantage of the many, many repositories available here on Github. Specifically searching on <a href="http://github.com" rel="nofollow noopener">github.com</a> for “quiz game python”, a few came up right away:</p>
<p><a href="https://github.com/cherryWood55/Quiz-Game" rel="nofollow noopener">https://github.com/cherryWood55/Quiz-Game</a></p>
<p><a href="https://github.com/abhijitnathwani/PyQuiz" rel="nofollow noopener">https://github.com/abhijitnathwani/PyQuiz</a></p>
<p><a href="https://github.com/sourcedexter/FlashCardQuiz" rel="nofollow noopener">https://github.com/sourcedexter/FlashCardQuiz</a></p>
<p><a href="https://github.com/raw-at/Q-A-Python-Game-GUI" rel="nofollow noopener">https://github.com/raw-at/Q-A-Python-Game-GUI</a></p>
<p><a href="https://github.com/harshnsdalai/Quizzard" rel="nofollow noopener">https://github.com/harshnsdalai/Quizzard</a></p>
<p>You might just take FlashCardQuiz, write up some questions/answers as per their sample file, and see if it meets your needs “as-is”.</p>
<p>For the future goal of using images, you’ll want to look into a GUI program, e.g. PyQuiz or Q-A-Python-Game-GUI.</p>

**Comment by GavynBryan** [comment]
<p>In addition to what fire-eggs is saying, it sounds like you’d want to make a web app for these “quizzes” and I agree that either Python or C#/ASP.NET are good choices for this. But, I’d recommend learning Python first if you don’t have any programming experience.</p>
<p>Then you’ll have a choice between lots of Python libraries for web development. (Like Django and Flask)</p>
<p>Make a few “console apps” first though. They’re going to be ugly terminal-based quizzes but you’ll learn how to implement algorithms and data structures.</p>

**Comment by Flamingo-verse** [comment]
<p>Thanks for the reply, and appolgies for my slow one in return.</p>
<p>Yes a desktop program was what I had in mind, with that said I also posted this question in Reddit and someone suggested making a website might be better. I’ve had more people say Python though, so I think that’ll be where I begin my Bilbo Baggins journey into the world of coding. </p>
<p>Thanks for the help, and the links… And also for reminding me about “flashcard”, that key word in searching as made a big difference, it’s one of those things that I just didn’t think of. Furthermore someone suggested the flashcard program Anki, which is apparently widely used by medical students, yet as a VetTech student I’ve never heard of it, it has pretty much what I need but I think I’m still going to set myself the goal of learning Python</p>
<p>Again, thanks for the help, it’s been truley appreciated! </p>

**Comment by Flamingo-verse** [comment]
<p>Thanks for taking the time to share some of your knowledge <a class="mention" href="/gavynbryan">@gavynbryan</a> </p>
<p>While I’ve found some programs which more or less fullfil my needs, I’ll still be giving it a go to try and learn coding, looks like Python will be where take my first baby steps.</p>
<p>Thanks for the information and pointing me in the right direction :slight_smile:</p>
